### PR TITLE
Feature/improved observability

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

 Improvements for shard synchronization

* Force sync-to-disk after shard synchronization.
* Improved progress messages for shard synchronization.
* Added configuration option `--rocksdb.sync-delay-threshold`.
  This option can be used to track if any RocksDB WAL sync operations is
  delayed by more than the configured value (in milliseconds). The intention
  is to get aware of severely delayed WAL sync operations.
* Improved error messages for cluster commit/abort operations.

- [x] :hankey: Bugfix 
- [ ] :pizza: New feature 
- [ ] :hammer: Refactoring 
- [ ] :book: CHANGELOG entry made
- [x] :muscle: The behavior in this PR was *manually tested*
- [ ] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [ ] No backports required
- [x] Backports required for: *3.7, 3.6*

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12139/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12150/

File: arangod/RocksDBEngine/RocksDBEngine.cpp [hunk 1]
```diff
@@ -1680,25 +1708,28 @@ std::vector<std::string> RocksDBEngine::currentWalFiles() const {
...
   if (_syncThread) {
     // _syncThread may be a nullptr, in case automatic syncing is turned off
-    _syncThread->syncWal();
+    res = _syncThread->syncWal();
   }
 
-  if (waitForCollector) {
+  if (res.ok() && waitForCollector) {
     rocksdb::FlushOptions flushOptions;
     flushOptions.wait = waitForSync;
```
Comment (danielhlarkin) [inline arangod/RocksDBEngine/RocksDBEngine.cpp:1721]: It looks like we typically call this function as `flushWal(true, false)`. Except we don't use the first parameter unless the second one is true. Odd.

File: arangod/RocksDBEngine/RocksDBSyncThread.cpp [hunk 2]
```diff
@@ -123,23 +131,45 @@ void RocksDBSyncThread::run() {
...
+          // still update our lastSyncTime to now, so we don't run into warnings
+          // later with syncs being reported as delayed
+          _lastSyncTime = lastSyncTime;
           continue;
         }
+      }
 
-        _lastSequenceNumber = lastSequenceNumber;
+      {
+        std::chrono::duration<double> diff = lastSyncTime - previousLastSyncTime;
```
Comment (danielhlarkin) [inline arangod/RocksDBEngine/RocksDBSyncThread.cpp:147]:
```suggestion
        std::chrono::duration<milliseconds> diff = lastSyncTime - previousLastSyncTime;
```

File: arangod/RocksDBEngine/RocksDBSyncThread.cpp [hunk 3]
```diff
@@ -123,23 +131,45 @@ void RocksDBSyncThread::run() {
...
+        if (_delayThreshold.count() > 0 && diff > _delayThreshold) {
+          LOG_TOPIC("5b708", INFO, Logger::ENGINES)
+            << "last RocksDB WAL sync happened longer ago than configured threshold. "
+            << "last sync happened " << std::chrono::duration_cast<std::chrono::milliseconds>(diff).count() << " ms ago, "
```
Comment (danielhlarkin) [inline arangod/RocksDBEngine/RocksDBSyncThread.cpp:151]:
```suggestion
            << "last sync happened " << diff.count() << " ms ago, "
```

**Review by danielhlarkin — CHANGES_REQUESTED** [review]

**Review by danielhlarkin — APPROVED** [review]

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12155/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**